eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -3,93 +3,373 @@
+  [string] $OpensourceApiApplicationSecret,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [Parameter(ParameterSetName = 'Interactive')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $TenantId,
+
+  [Parameter(ParameterSetName = 'Provisioner', Mandatory = $true)]
+  [Parameter(ParameterSetName = 'Interactive')]
+  [ValidatePattern('^[0-9a-f]{8}(-[0-9a-f]{4}){3}-[0-9a-f]{12}$')]
+  [string] $SubscriptionId,
+
+  [Parameter(ParameterSetName = 'Provisioner')]
+  [string] $GithubAliasCachePath,
+
+  [Parameter()]
+  [ValidateNotNullOrEmpty()]
+  [string] $Environment = "AzureCloud",
+
+  [Parameter()]
+  [switch] $DeleteNonCompliantGroups,
 
-    [Parameter()]
-    [ValidateNotNullOrEmpty()]
-    $Environment = "AzureCloud",
+  [Parameter()]
+  [int] $DeleteAfterHours = 24,
 
-    [Parameter()]
-    [switch] $Force,
+  [Parameter()]
+  [string] $AllowListPath = "$PSScriptRoot/cleanup-allowlist",
 
-    [Parameter(ValueFromRemainingArguments = $true)]
-    $IgnoreUnusedArguments
+  [Parameter()]
+  [switch] $Force,
+
+  [Parameter(ParameterSetName = 'Interactive')]
+  [switch] $Login,
+
+  [Parameter(ValueFromRemainingArguments = $true)]
+  $IgnoreUnusedArguments
 )
 
-&"$PSScriptRoot/../common/scripts/Import-AzModules.ps1"
+Set-StrictMode -Version 3
 
 # Import resource management helpers and override its Log function.
-. "$PSScriptRoot\..\common\scripts\Helpers\Resource-Helpers.ps1"
+. "$PSScriptRoot/../common/scripts/Helpers/Resource-Helpers.ps1"
+# Import helpers for querying repos.opensource.microsoft.com API
+. "$PSScriptRoot/../common/scripts/Helpers/Metadata-Helpers.ps1"
+
+$OwnerAliasCache = @{}
+$Exceptions = [System.Collections.Generic.HashSet[String]]@()
+$ShouldCheckAllowList = Test-Path $AllowListPath
+$IsProvisionerApp = $PSCmdlet.ParameterSetName -eq "Provisioner"
 
 function Log($Message) {
   Write-Host $Message
 }
 
+function IsValidAlias
+{
+  param(
+    [Parameter(Mandatory = $true)]
+    [string]$Alias
+  )
+
+  if ($OwnerAliasCache.ContainsKey($Alias)) {
+    return $OwnerAliasCache[$Alias]
+  }
+
+  # AAD apps require a higher level of permission requiring admin consent to query the MS Graph list API
+  # https://docs.microsoft.com/en-us/graph/api/user-list?view=graph-rest-1.0&tabs=http#permissions
+  # The Get-AzAdUser call uses the list API under the hood (`/users/$filter=<alias>`)
+  # and for some reason the Get API (`/user/<id or user principal name>`) also returns 401
+  # with User.Read and User.ReadBasic.All permissions when called with an AAD app.
+  # For this reason, skip trying to query MS Graph directly in provisioner mode.
+  # The owner alias cache should already be pre-populated with all user records from the
+  # github -> ms alias mapping retrieved via the repos.opensource.microsoft.com API, however
+  # this will not include any security groups, in the case an owner tag does not contain
+  # individual user aliases.
+  if ($IsProvisionerApp) {
```

Comment:
Given our new approach do we even want to keep this code?

Reply:
We could remove it and it would certainly make things a little simpler, though I also like that it provides a convenient way to see what the delta is between valid aliases and github aliases if you run locally.

Reply:
OK I'm fine with keeping it even if it only works locally.